diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2fa30b0..f828d9c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,51 +3,51 @@
 # found in the top-level directory of this distribution.
 
 set(detail_header
-        ../include/cppast/detail/assert.hpp
-        ../include/cppast/detail/intrusive_list.hpp)
+        ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/detail/assert.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/detail/intrusive_list.hpp)
 set(header
-    ../include/cppast/code_generator.hpp
-    ../include/cppast/compile_config.hpp
-    ../include/cppast/cpp_alias_template.hpp
-    ../include/cppast/cpp_array_type.hpp
-    ../include/cppast/cpp_attribute.hpp
-    ../include/cppast/cpp_class.hpp
-    ../include/cppast/cpp_class_template.hpp
-    ../include/cppast/cpp_decltype_type.hpp
-    ../include/cppast/cpp_entity.hpp
-    ../include/cppast/cpp_entity_container.hpp
-    ../include/cppast/cpp_entity_index.hpp
-    ../include/cppast/cpp_entity_kind.hpp
-    ../include/cppast/cpp_entity_ref.hpp
-    ../include/cppast/cpp_enum.hpp
-    ../include/cppast/cpp_expression.hpp
-    ../include/cppast/cpp_file.hpp
-    ../include/cppast/cpp_forward_declarable.hpp
-    ../include/cppast/cpp_friend.hpp
-    ../include/cppast/cpp_function.hpp
-    ../include/cppast/cpp_function_template.hpp
-    ../include/cppast/cpp_function_type.hpp
-    ../include/cppast/cpp_language_linkage.hpp
-    ../include/cppast/cpp_member_function.hpp
-    ../include/cppast/cpp_member_variable.hpp
-    ../include/cppast/cpp_namespace.hpp
-    ../include/cppast/cpp_preprocessor.hpp
-    ../include/cppast/cpp_static_assert.hpp
-    ../include/cppast/cpp_storage_class_specifiers.hpp
-    ../include/cppast/cpp_template.hpp
-    ../include/cppast/cpp_template_parameter.hpp
-    ../include/cppast/cpp_token.hpp
-    ../include/cppast/cpp_type.hpp
-    ../include/cppast/cpp_type_alias.hpp
-    ../include/cppast/cpp_variable.hpp
-    ../include/cppast/cpp_variable_base.hpp
-    ../include/cppast/cpp_variable_template.hpp
-    ../include/cppast/diagnostic.hpp
-    ../include/cppast/diagnostic_logger.hpp
-    ../include/cppast/cppast_fwd.hpp
-    ../include/cppast/libclang_parser.hpp
-    ../include/cppast/parser.hpp
-    ../include/cppast/visitor.hpp)
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/code_generator.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/compile_config.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_alias_template.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_array_type.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_attribute.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_class.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_class_template.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_decltype_type.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_entity.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_entity_container.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_entity_index.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_entity_kind.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_entity_ref.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_enum.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_expression.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_file.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_forward_declarable.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_friend.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_function.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_function_template.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_function_type.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_language_linkage.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_member_function.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_member_variable.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_namespace.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_preprocessor.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_static_assert.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_storage_class_specifiers.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_template.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_template_parameter.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_token.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_type.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_type_alias.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_variable.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_variable_base.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cpp_variable_template.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/diagnostic.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/diagnostic_logger.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/cppast_fwd.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/libclang_parser.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/parser.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include/cppast/visitor.hpp)
 set(source
         code_generator.cpp
         cpp_alias_template.cpp
@@ -103,8 +103,14 @@ set(libclang_source
         libclang/variable_parser.cpp)
 
 add_library(cppast ${detail_header} ${header} ${source} ${libclang_source})
+set(cppast_public_headers ${detail_header})
+list(APPEND cppast_public_headers ${header})
+set_target_properties(cppast PROPERTIES PUBLIC_HEADER "${cppast_public_headers}")
 target_compile_features(cppast PUBLIC cxx_std_11)
-target_include_directories(cppast PRIVATE ../include SYSTEM INTERFACE ../include)
+target_include_directories(cppast PRIVATE ../include SYSTEM INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+    $<INSTALL_INTERFACE:include>
+)
 target_link_libraries(cppast PUBLIC type_safe _cppast_tiny_process _cppast_libclang)
 target_compile_definitions(cppast PUBLIC
                                 CPPAST_VERSION_MINOR="${cppast_VERSION_MINOR}"
